fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/impl/dimension/FabricDimensionClientInit.java

```diff
@@ -0,0 +1,25 @@
+package net.fabricmc.fabric.impl.dimension;
+
+import net.fabricmc.fabric.api.network.ClientSidePacketRegistry;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.text.LiteralText;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Client entry point for fabric-dimensions
+ */
+public final class FabricDimensionClientInit {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public static void onClientInit() {
+        ClientSidePacketRegistry.INSTANCE.register(DimensionIdsFixer.ID, (ctx, buf) -> DimensionIdsFixer.receivePacket(ctx, buf, (e) -> {
+            LOGGER.error("Dimension id remapping failed!", e);
+            MinecraftClient.getInstance().execute(() -> ((ClientPlayerEntity) ctx.getPlayer()).networkHandler.getConnection().disconnect(
+                    new LiteralText("Dimension id remapping failed: " + e.getMessage())
```

Comment:
Throwable.getMessage() is sometimes blank (e.g. jvm generated NPE), logging/sending the Throwable itself (-> Throwable.toString()) adds both the exception type and message (if present).

I think it'd be nicer to throw and catch here instead of this exception handler funnelling.

DimensionIdsFixer.receivePacket schedules on the main thread while blocking the network thread until done, which is not a good idea and may leave the network thread stuck if the main thread stops executing scheduled tasks, besides potentially causing unnecessary delay. This should be doable by pushing processing to the main thread exactly once.

Reply:
The whole synchronization code is actually taken from fabric-registry-sync. I agree, this should probably be simplified. I reckon DimensionIdsFixer.receivePacket may entirely be inlined here, removing the need for those hoops.